packages/integration_test/lib/integration_test_driver.dart

```diff
@@ -24,23 +24,29 @@ String testOutputsDirectory =
 typedef ResponseDataCallback = FutureOr<void> Function(Map<String, dynamic>?);
 
 /// Writes a json-serializable data to
-/// [testOutputsDirectory]/`testOutputFilename.json`.
+/// [testOutputsDirectory]/`testOutputFilename_<key>.json`, where key is the
+/// name provided as the report key for the response data.
 ///
 /// This is the default `responseDataCallback` in [integrationDriver].
 Future<void> writeResponseData(
   Map<String, dynamic>? data, {
   String testOutputFilename = 'integration_response_data',
   String? destinationDirectory,
 }) async {
+  if (data == null) {
+    return;
+  }
   assert(testOutputFilename != null);
   destinationDirectory ??= testOutputsDirectory;
   await fs.directory(destinationDirectory).create(recursive: true);
-  final File file = fs.file(path.join(
-    destinationDirectory,
-    '$testOutputFilename.json',
-  ));
-  final String resultString = _encodeJson(data, true);
-  await file.writeAsString(resultString);
+  for (final String key in data.keys.where((String key) => key != 'screenshots')) {
+    final File file = fs.file(path.join(
+      destinationDirectory,
+      '${testOutputFilename}_$key.json',
```

Comment:
Just a drive by question, will the file names match the name of the test?

Reply:
They'll match whatever was given as the 'reportKey' argument. In the original bug, you'd get `integration_response_data_ favorites_operations_timeline.json` etc.

Reply:
(See the test below for sample file names)